src/pokemon.c

```diff
@@ -6941,3 +6942,180 @@ void UpdateDaysPassedSinceFormChange(u16 days)
+    }
+    else if (move == MOVE_TERA_BLAST && GetActiveGimmick(battler) == GIMMICK_TERA && gBattleMons[battler].species == species)
+    {
+        return GetMonData(mon, MON_DATA_TERA_TYPE);
+    }
+    else if (move == MOVE_TERA_STARSTORM && species == SPECIES_TERAPAGOS_STELLAR)
+    {
+        return TYPE_STELLAR;
+    }
+    else if (move == MOVE_HIDDEN_POWER)
+    {
+        return CalculateHiddenPowerType(mon);
+    }
+    else if (move == MOVE_AURA_WHEEL && species == SPECIES_MORPEKO_HANGRY)
+    {
+        type = TYPE_DARK;
+    }
+    else if (gMovesInfo[move].effect == EFFECT_CHANGE_TYPE_ON_ITEM && heldItemEffect == gMovesInfo[move].argument)
+    {
+        return ItemId_GetSecondaryId(heldItem);
+    }
+    else if (move == MOVE_NATURAL_GIFT && ItemId_GetPocket(heldItem) == POCKET_BERRIES)
+    {
+        return gNaturalGiftTable[ITEM_TO_BERRY(heldItem)].type;
+    }
+    else if (move == MOVE_RAGING_BULL 
+        && (species == SPECIES_TAUROS_PALDEAN_COMBAT_BREED 
+        || species == SPECIES_TAUROS_PALDEAN_BLAZE_BREED 
+        || species == SPECIES_TAUROS_PALDEAN_AQUA_BREED))
+    {
+        return type2;
+    }        
+    else if (move == MOVE_REVELATION_DANCE)
+    {
+        if (gBattleMons[battler].species != species && type1 != TYPE_MYSTERY)
+        {
+            type = type1;
+        }
+        else if (gBattleMons[battler].species != species && type2 != TYPE_MYSTERY)
+        {
+            type = type2;
+        }
+        else if (GetBattlerTeraType(battler) != TYPE_STELLAR 
+            && (GetActiveGimmick(battler) == GIMMICK_TERA || IsGimmickSelected(battler, GIMMICK_TERA)))
+        {
+            type = GetMonData(mon, MON_DATA_TERA_TYPE);
+        }        
+        else if (gBattleMons[battler].type1 != TYPE_MYSTERY)
+        {
+            type = gBattleMons[battler].type1;
+        }
+        else if (gBattleMons[battler].type2 != TYPE_MYSTERY)
+        {
+            type = gBattleMons[battler].type2;
+        }
+        else if (gBattleMons[battler].type3 != TYPE_MYSTERY)
+        {
+            type = gBattleMons[battler].type3;
+        }
+    }
+    else if (gMovesInfo[move].effect == EFFECT_TERRAIN_PULSE 
+          && ((IsMonGrounded(heldItemEffect, ability, type1, type2) && gBattleMons[battler].species != species) 
+           || (IsBattlerTerrainAffected(battler, STATUS_FIELD_TERRAIN_ANY) && gBattleMons[battler].species == species)))
+    {  
+        if (gFieldStatuses & STATUS_FIELD_ELECTRIC_TERRAIN)
+        {
+            return TYPE_ELECTRIC;
+        }
+        else if (gFieldStatuses & STATUS_FIELD_GRASSY_TERRAIN)
+        {
+            return TYPE_GRASS;
+        }
+        else if (gFieldStatuses & STATUS_FIELD_MISTY_TERRAIN)
+        {
+            return TYPE_FAIRY;
+        }
+        else if (gFieldStatuses & STATUS_FIELD_PSYCHIC_TERRAIN)
+        {
+            return TYPE_PSYCHIC;
+        }
+        else //failsafe
+        {
+            type = TYPE_NORMAL;
+        }        
```

Comment:
I would remove the brackets here since each statement has only one line in this if-else block.